HBASE-28647 Support streams in REST Client, RemoteHTable and RemoteAdmin

stoty opened the pull request.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 38s |  master passed  |
| +1 :green_heart: |  compile  |   0m 18s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 13s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 15s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 30s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 18s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 18s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 10s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 14s |  hbase-rest generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0)  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 27s |  hbase-rest in the patch passed.  |
|  |   |  21m 44s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 518782e13d9d 5.4.0-182-generic #202-Ubuntu SMP Fri Apr 26 12:29:36 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 52eef65d37 |
| Default Java | Temurin-1.8.0_352-b08 |
| javadoc | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/artifact/yetus-jdk8-hadoop3-check/output/diff-javadoc-javadoc-hbase-rest.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/testReport/ |
| Max. process+thread count | 1605 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 52s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 43s |  master passed  |
| +1 :green_heart: |  compile  |   0m 22s |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m 36s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 17s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 52s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 20s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 20s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 35s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 15s |  hbase-rest generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0)  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 42s |  hbase-rest in the patch passed.  |
|  |   |  25m 37s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 50c5eaac5b15 5.4.0-174-generic #193-Ubuntu SMP Thu Mar 7 14:29:28 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 52eef65d37 |
| Default Java | Eclipse Adoptium-17.0.10+7 |
| javadoc | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/artifact/yetus-jdk17-hadoop3-check/output/diff-javadoc-javadoc-hbase-rest.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/testReport/ |
| Max. process+thread count | 1618 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 41s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 49s |  master passed  |
| +1 :green_heart: |  compile  |   0m 33s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 53s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 36s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 37s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 34s |  the patch passed  |
| -0 :warning: |  javac  |   0m 34s |  hbase-rest generated 3 new + 185 unchanged - 0 fixed = 188 total (was 185)  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |   6m 54s |  Patch does not cause any errors with Hadoop 3.3.6.  |
| +1 :green_heart: |  spotless  |   0m 54s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 45s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  The patch does not generate ASF License warnings.  |
|  |   |  28m 17s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux b3e7fd76ad1b 5.4.0-182-generic #202-Ubuntu SMP Fri Apr 26 12:29:36 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 52eef65d37 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| javac | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/artifact/yetus-general-check/output/diff-compile-javac-hbase-rest.txt |
| Max. process+thread count | 78 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 43s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 24s |  master passed  |
| +1 :green_heart: |  compile  |   0m 31s |  master passed  |
| +1 :green_heart: |  shadedjars  |   8m 16s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 12s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 25s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 25s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m 47s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 22s |  hbase-rest generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0)  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   6m 32s |  hbase-rest in the patch passed.  |
|  |   |  35m  2s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 6a6f1ff50fe0 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 52eef65d37 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| javadoc | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/artifact/yetus-jdk11-hadoop3-check/output/diff-javadoc-javadoc-hbase-rest.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/testReport/ |
| Max. process+thread count | 1655 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache9 commented on hbase-rest/src/main/java/org/apache/hadoop/hbase/rest/ProtobufHttpEntity.java:
```diff
@@ -0,0 +1,87 @@
+ * For sending a Protobuf encoded object via Apache HttpClient efficiently without an interim byte
+ * array. This exposes the underlying Apache HttpClient types, but so do the other client classes.
+ */
+@InterfaceAudience.Public
```
Does this need to be IA.Public?

stoty replied:
Yes, we expect the user to explicitly create these entities.

i have not updated the tests to use them, but the intentended usage is similar to how 
org.apache.hadoop.hbase.rest.client.RemoteHTable.put(Put) is implemented.

Apache9 replied:
At least have a section in the javadoc to show how to use this class?

And since we have not reach an agreement on whether to move RemoteAdmin and RemoteHTable to src/main, I'm not sure whether we should make this class IA.Public...

stoty replied:
I only mentined RemoHTable as example where this is used, but this 
is not limited to RemotHTable, it should also be used directly in user code.

I will add a standalone test case that uses this class as an illustration.

stoty replied:
> At least have a section in the javadoc to show how to use this class?
> 
> And since we have not reach an agreement on whether to move RemoteAdmin and RemoteHTable to src/main, I'm not sure whether we should make this class IA.Public...

This is not tied to RemoteAdmin / RemoteHTable. 
This is just a more efficient way to use the Client class (or a custom client based on Apache HttpClient)

The RemoteAdmin / RemoteHTable  API does not even change, this new class only used in the RemoteHTable implementation.

Apache9 commented on hbase-rest/src/main/java/org/apache/hadoop/hbase/rest/ProtobufHttpEntity.java:
```diff
@@ -0,0 +1,87 @@
+
+  @Override
+  public boolean isStreaming() {
+    // TODO Auto-generated method stub
```
Remove this line?

stoty replied:
Sure

Apache9 reviewed (commented):
I do not fully understand what does the streams mean here...

All requests and responses are fully kept in memory here I think?

stoty commented:
> I do not fully understand what does the streams mean here...
> 
> All requests and responses are fully kept in memory here I think?

Currently they are.
The current code calls the Apache HttpClient getResponseBody() method, which will cause the client to wait untill all data is received, and load it into a byte array.

However, the goal is to avoid having to do that.

Protobuf primarily works on streams, so for a large resultset, we may reduce both processing (wall clock) time and memory consumption by not buffering the whole response into memory, but reading directly from the stream, so that
* We do not have to wait for the full response to arrive before starting to process it.
* We do not have to copy the whole response into a single byte array.
* The processed response segments can be GCd while we are processing the rest of the message.

The Cell/Cellset structures are still kept in memory, but we avoid having to explicitly store them twice during processing (once the serialized byte array and once the java POJOs)

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 38s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 58s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 35s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 33s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 45s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 48s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 33s |  |  the patch passed  |
| -0 :warning: |  javac  |   0m 33s | [/results-compile-javac-hbase-rest.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/2/artifact/yetus-general-check/output/results-compile-javac-hbase-rest.txt) |  hbase-rest generated 3 new + 183 unchanged - 0 fixed = 186 total (was 183)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   0m 38s |  |  the patch passed  |
| -1 :x: |  hadoopcheck  |   5m 48s |  |  The patch causes 10 errors with Hadoop v3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 43s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 11s |  |  The patch does not generate ASF License warnings.  |
|  |   |  17m 57s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 67709cb57af2 5.4.0-177-generic #197-Ubuntu SMP Thu Mar 28 22:45:47 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 4fd2161b57c777f37d9380ffa62fddedb5b0d921 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| hadoopcheck | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/2/artifact/yetus-general-check/output/patch-javac-3.4.0.txt |
| Max. process+thread count | 85 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/2/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 42s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 16s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 28s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 22s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m 21s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 14s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 23s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 23s |  |  the patch passed  |
| -0 :warning: |  javadoc  |   0m 15s | [/results-javadoc-javadoc-hbase-rest.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/2/artifact/yetus-jdk17-hadoop3-check/output/results-javadoc-javadoc-hbase-rest.txt) |  hbase-rest generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0)  |
| +1 :green_heart: |  shadedjars  |   6m 23s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 12s |  |  hbase-rest in the patch passed.  |
|  |   |  28m 38s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/2/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux be0b3b85ee52 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 4fd2161b57c777f37d9380ffa62fddedb5b0d921 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/2/testReport/ |
| Max. process+thread count | 1520 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/2/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 25s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  6s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 18s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 15s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 57s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 37s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 20s |  |  the patch passed  |
| -0 :warning: |  javadoc  |   0m 13s | [/results-javadoc-javadoc-hbase-rest.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/3/artifact/yetus-jdk17-hadoop3-check/output/results-javadoc-javadoc-hbase-rest.txt) |  hbase-rest generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0)  |
| +1 :green_heart: |  shadedjars  |   6m 30s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 16s |  |  hbase-rest in the patch passed.  |
|  |   |  26m  4s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/3/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux aaabde1c74bc 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 05adfadaf0acd5a2ebe7e794564aef1d5c7fb142 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/3/testReport/ |
| Max. process+thread count | 1672 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/3/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 27s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 25s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 56s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 15s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 43s |  |  master passed  |
| +1 :green_heart: |  spotless  |   1m 16s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  9s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 44s |  |  the patch passed  |
| -0 :warning: |  javac  |   0m 44s | [/results-compile-javac-hbase-rest.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/3/artifact/yetus-general-check/output/results-compile-javac-hbase-rest.txt) |  hbase-rest generated 3 new + 183 unchanged - 0 fixed = 186 total (was 183)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   0m 52s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  14m  0s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 55s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 12s |  |  The patch does not generate ASF License warnings.  |
|  |   |  36m 31s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 99e1cee57171 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 05adfadaf0acd5a2ebe7e794564aef1d5c7fb142 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 82 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/3/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache9 commented:
Please fix the javac and javadoc warnings if possible?

stoty commented:
I have added org.apache.hadoop.hbase.rest.TestGetAndPutResource.testMultipleCellPutPBEntity() to illustrate how ProtobufEntity can be used to more efficiently execute PUT operations.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   4m 10s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  2s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 21s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 17s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 21s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 58s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 22s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 22s |  |  the patch passed  |
| -0 :warning: |  javadoc  |   0m 16s | [/results-javadoc-javadoc-hbase-rest.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/4/artifact/yetus-jdk17-hadoop3-check/output/results-javadoc-javadoc-hbase-rest.txt) |  hbase-rest generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0)  |
| +1 :green_heart: |  shadedjars  |   5m 20s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 23s |  |  hbase-rest in the patch passed.  |
|  |   |  26m 37s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/4/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux a699f0bc9189 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / c61d318f4887944d8bd02a05365b410e8f1e0310 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/4/testReport/ |
| Max. process+thread count | 1654 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/4/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 36s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  0s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 36s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 31s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 46s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 58s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 35s |  |  the patch passed  |
| -0 :warning: |  javac  |   0m 35s | [/results-compile-javac-hbase-rest.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/4/artifact/yetus-general-check/output/results-compile-javac-hbase-rest.txt) |  hbase-rest generated 3 new + 183 unchanged - 0 fixed = 186 total (was 183)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 10s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   0m 38s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 36s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 11s |  |  The patch does not generate ASF License warnings.  |
|  |   |  28m 14s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 0d95c9721af7 5.4.0-177-generic #197-Ubuntu SMP Thu Mar 28 22:45:47 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / c61d318f4887944d8bd02a05365b410e8f1e0310 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/4/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 25s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 12s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 18s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 15s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 48s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  5s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 13s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 51s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 42s |  |  hbase-rest in the patch passed.  |
|  |   |  25m 13s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/5/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 185c2f093f33 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3f2010be80c941c9ebcb403ea977085d08d3b4d8 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/5/testReport/ |
| Max. process+thread count | 1652 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/5/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 37s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 59s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 34s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 31s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 55s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 34s |  |  the patch passed  |
| -0 :warning: |  javac  |   0m 34s | [/results-compile-javac-hbase-rest.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/5/artifact/yetus-general-check/output/results-compile-javac-hbase-rest.txt) |  hbase-rest generated 1 new + 183 unchanged - 0 fixed = 184 total (was 183)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 11s | [/results-checkstyle-hbase-rest.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/5/artifact/yetus-general-check/output/results-checkstyle-hbase-rest.txt) |  hbase-rest: The patch generated 1 new + 2 unchanged - 0 fixed = 3 total (was 2)  |
| +1 :green_heart: |  spotbugs  |   0m 39s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 34s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   |  27m 59s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/5/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 5351a125f6b7 5.4.0-177-generic #197-Ubuntu SMP Thu Mar 28 22:45:47 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3f2010be80c941c9ebcb403ea977085d08d3b4d8 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 84 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/5/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 37s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 58s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 22s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 16s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 19s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 59s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 22s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 22s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 16s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 20s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 25s |  |  hbase-rest in the patch passed.  |
|  |   |  22m 56s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/6/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 4d774a6b3483 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 1ccfecb2545b91d0110f33c08868e63d1192e1d0 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/6/testReport/ |
| Max. process+thread count | 1680 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/6/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 38s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 36s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 32s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 45s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 55s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 34s |  |  the patch passed  |
| -0 :warning: |  javac  |   0m 34s | [/results-compile-javac-hbase-rest.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/6/artifact/yetus-general-check/output/results-compile-javac-hbase-rest.txt) |  hbase-rest generated 1 new + 183 unchanged - 0 fixed = 184 total (was 183)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 11s | [/results-checkstyle-hbase-rest.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/6/artifact/yetus-general-check/output/results-checkstyle-hbase-rest.txt) |  hbase-rest: The patch generated 1 new + 2 unchanged - 0 fixed = 3 total (was 2)  |
| +1 :green_heart: |  spotbugs  |   0m 39s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 48s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 11s |  |  The patch does not generate ASF License warnings.  |
|  |   |  28m 25s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/6/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 4c70a47798e7 5.4.0-177-generic #197-Ubuntu SMP Thu Mar 28 22:45:47 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 1ccfecb2545b91d0110f33c08868e63d1192e1d0 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 83 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/6/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 26s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 19s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 20s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 15s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 47s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 19s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 19s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 14s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 48s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 40s |  |  hbase-rest in the patch passed.  |
|  |   |  24m 13s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/7/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 8cde5cfd046d 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3f9243d283f91c83b068a2ebb6d8b0e71bb5d103 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/7/testReport/ |
| Max. process+thread count | 1646 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/7/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 37s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 55s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 33s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 32s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 53s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 34s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 34s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   0m 39s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 26s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   |  27m 45s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/7/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux d770c264f85c 5.4.0-177-generic #197-Ubuntu SMP Thu Mar 28 22:45:47 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3f9243d283f91c83b068a2ebb6d8b0e71bb5d103 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 83 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/7/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

stoty commented:
Fixed the warnings (though most were in code that I haven't touched)
and added a test that can serve as an example on using ProtobufEntity.
PTAL @Apache9 .

stoty commented:
Now that you're back from vacation and hopefully settled can you take a look at this again, @Apache9 ?

stoty commented on hbase-rest/src/test/java/org/apache/hadoop/hbase/rest/TestGetAndPutResource.java:
```diff
@@ -167,6 +170,39 @@ public void testMultipleCellCheckPutPB() throws IOException {
+    path.append('/');
+    path.append("dummy_row");
+    HttpPut httpPut = new HttpPut(path.toString());
+    httpPut.setEntity(new ProtobufHttpEntity(model));
```
This is how a normal client can use ProtobufHttpEntity, @Apache9 .

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 39s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 34s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 32s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 45s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 48s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 33s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 33s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   0m 37s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 16s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 43s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 11s |  |  The patch does not generate ASF License warnings.  |
|  |   |  27m 19s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/8/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux beef180c5b0d 5.4.0-195-generic #215-Ubuntu SMP Fri Aug 2 18:28:05 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 5b0906dd8f4e05a6ea4af04bcfb3ec2b2473cf15 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 83 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/8/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 30s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 16s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 21s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 17s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 58s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  2s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 16s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 52s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 50s |  |  hbase-rest in the patch passed.  |
|  |   |  24m 43s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/9/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 85fb0039742f 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / a0803e61bd0eb0d5345ac008b5b080b99693e130 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/9/testReport/ |
| Max. process+thread count | 1672 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/9/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 31s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 17s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 33s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m  9s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 35s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 45s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 31s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 31s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m  9s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   0m 41s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m 42s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 43s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   |  30m  7s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/9/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6010 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux b9ac3c8ec6bf 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / a0803e61bd0eb0d5345ac008b5b080b99693e130 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 84 (vs. ulimit of 30000) |
| modules | C: hbase-rest U: hbase-rest |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6010/9/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.
